packages/shared/src/components/modals/referral/GenericReferralModal.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { ReactElement, useState } from 'react';
+import { Modal, ModalProps } from '../common/Modal';
+import { cloudinary } from '../../../lib/image';
+import CloseButton from '../../CloseButton';
+import { ModalSize } from '../common/types';
+import { Button, ButtonSize } from '../../buttons/Button';
+import { TextField } from '../../fields/TextField';
+import {
+  getFacebookShareLink,
+  getTwitterShareLink,
+  getWhatsappShareLink,
+} from '../../../lib/share';
+import WhatsappIcon from '../../icons/Whatsapp';
+import { SimpleTooltip } from '../../tooltips';
+import FacebookIcon from '../../icons/Facebook';
+import TwitterIcon from '../../icons/Twitter';
+import { useShareOrCopyLink } from '../../../hooks/useShareOrCopyLink';
+import { link } from '../../../lib/links';
+import { labels } from '../../../lib';
+import { AnalyticsEvent, TargetId } from '../../../lib/analytics';
+import { ReferralCampaignKey, useReferralCampaign } from '../../../hooks';
+
+function GenericReferralModal({
+  onRequestClose,
+  ...props
+}: ModalProps): ReactElement {
+  const [shareState, setShareState] = useState(false);
+  const { url } = useReferralCampaign({
+    campaignKey: ReferralCampaignKey.Generic,
+  });
+  const inviteLink = url || link.referral.defaultUrl;
+  const [copyingLink, onShareOrCopyLink] = useShareOrCopyLink({
+    text: labels.referral.generic.inviteText,
+    link: inviteLink,
+    trackObject: () => ({
+      event_name: AnalyticsEvent.CopyReferralLink,
+      target_id: TargetId.GenericReferralPopup,
+    }),
+  });
+  const onShareClick = () => {
+    onShareOrCopyLink();
+    setShareState(true);
+  };
+
+  return (
+    <Modal {...props} onRequestClose={onRequestClose} size={ModalSize.Small}>
+      <CloseButton
+        onClick={onRequestClose}
+        position="absolute"
+        buttonSize={ButtonSize.Small}
+        className="top-4 right-4 z-2 !btn-secondary"
+      />
+      <Modal.Body>
+        <div className="flex relative z-1 flex-col justify-end items-center mb-5 w-full aspect-square">
+          <h1 className="mb-4 font-bold typo-mega3">Invite friends</h1>
+          <p className="typo-title3 text-theme-label-secondary">
+            And make Ido (our CTO) smile again.
+          </p>
+        </div>
+        <img
+          src={
+            !shareState
+              ? cloudinary.referralCampaign.generic.sad
+              : cloudinary.referralCampaign.generic.happy
+          }
```

Comment:
Just wondering if it could be done like this
```suggestion
          src={
            cloudinary.referralCampaign.generic[!shareState ? 'sad' : 'happy']
          }
```

Reply:
Should yes, let me try :)